lib/dgram.js

```diff
@@ -272,6 +267,65 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+function fixBuffer(buffer) {
+  var length = 0;
+
+  for (var i = 0, l = buffer.length; i < l; i++) {
+    if (typeof buffer[i] === 'string')
+      buffer[i] = new Buffer(buffer[i]);
+    else if (!(buffer[i] instanceof Buffer))
+      return -1;
+
+    length += buffer[i].length;
+  }
+
+  return length;
+}
+
+function enqueue(self, toEnqueue) {
+  // If the send queue hasn't been initialized yet, do it, and install an
+  // event handler that flushes the send queue after binding is done.
+  if (!self._sendQueue) {
+    self._sendQueue = [];
+    self.once('listening', function() {
+      // Flush the send queue.
+      for (var i = 0; i < self._sendQueue.length; i++)
+        self.send.apply(self, self._sendQueue[i]);
+      self._sendQueue = undefined;
+    });
+  }
+  self._sendQueue.push(toEnqueue);
+  return;
+}
+
+Socket.prototype.send = function(buffer,
+                                 offset,
+                                 length,
+                                 port,
+                                 address,
+                                 callback) {
+  var self = this;
+
+  if (arguments.length === 5 || arguments.length === 6) {
+    buffer = sliceBuffer(buffer, offset, length);
+  }
+
+  if (arguments.length < 5) {
+    callback = port;
+    port = offset;
+    address = length;
+  }
+
+  if (!Array.isArray(buffer))
+    buffer = [ buffer ];
```

Comment:
It's cheaper to check for `instanceof Buffer` (or `HasInstance()` from C++) than to always wrap the buffer in an array.